doc/api/util.md

```diff
@@ -325,6 +325,60 @@ The `--throw-deprecation` command-line flag and `process.throwDeprecation`
 property take precedence over `--trace-deprecation` and
 `process.traceDeprecation`.
 
+## `util.diff(actual, expected)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `actual` {any}
+* `expected` {any}
+* Returns: {string | null} A string if there is difference
+  between `actual` and `expected`, `null` otherwise
+
+[`util.diff()`][] is a utility function that compares two values and returns a
+string that represents the difference between them. It is useful for debugging
+and testing.
+The output is exactly the same as the one used by all the assertion errors, therefore
+it is internally using the Myer's diff algorithm to compute the differences.
+
+```js
+const { diff } = require('node:util');
+
+console.log(diff({ a: 1, b: 2 }, { a: 1, b: 3 }));
+// above will log:
+/*
++ actual - expected
+
+  {
+    a: 1,
++   b: 2
+-   b: 3
+  }
+*/
+
+// if the sum of the strings length is less than 12 characters, a short diff will be returned
```

Comment:
nit: since we're exposing this method as a utility, would it make sense to also provide a way to configure its behavior?  
For example, enabling/disabling short diff or Mayers.

Reply:
I would actually rather not expose what we currently do internally. I don't think this is something that users would expect from a generic purpose diff tool :)
So just removing the behavior for the exposed one would be ideal in my opinion. If we add option later, great, but the default just be a plain diff.